source/tutorials/zfs.rst

```diff
@@ -0,0 +1,275 @@
+when you upgrade ZFS. Do not delete the source location when you have completed this tutorial. You will need it later.
+
+Compile the module
+==================
+
+We will build the module using DKMS. This will enable us to keep the module up to date as new kernels are released in the future.
+
+
+#. Check out the tagged version that you plan to use. As of the time of this
+   writing, the latest release tag is `0.8.4`:
+
+   .. code-block:: bash
+
+      cd /opt/src/zfs
+      git checkout 0.8.4
+
+#. Copy the source code into `/usr/src/zfs-0.8.4`. This exposes the source
+   code to DKMS. We will build the code from the new location:
+
+   .. code-block:: bash
+
+      sudo cp -Rv /opt/src/zfs /usr/src/zfs-0.8.4
+      cd /usr/src/zfs-0.8.4
+
+#. The ZFS distribution provides a script to build a suitable dkms.conf file.
+   Build dkms.conf and install it into the DKMS tree.
+
+   .. code-block:: bash
+
+      cd /usr/src/zfs-0.8.4
+      scripts/dkms.mkconf -n zfs -v 0.8.4 -f dkms.conf
+      sudo dkms add -m zfs -v 0.8.4
+      sudo dkms build -m zfs -v 0.8.4
+      sudo dkms install -m zfs -v 0.8.4
+
+#. Observe that this install the zfs kernel modules to: 
+   :file:`/usr/lib/modules/<kernel-name>/extra/zfs`
+
+   In addition, this installs the zfs userspace tools to:
+
+   .. code-block:: console
+
+      + /usr/local/
+      |--+ bin/
+         |--zvol_wait
+         |--zgenhostid
+         |--raidz_test
+      |--+ etc/
+         |--+ zfs/
+            |--* zed.d/
+            |--+ zpool.d/
+            |--zfs-functions
+      |--+ include/libzfs/ [contents omitted]
+      |--+ lib/
+      |--+ libexec/
+         |--+ zfs/
+            |-- zpool.d/
+            |-- zed.d
+      |--+ share/zfs/ [contents omitted]
+      |--+ sbin/
+         |--fsck.zfs
+         |--zpool
+         |--zdb
+         |--zed
+         |--zfs
+         |--zhack
+         |--zinject
+         |--zpool
+         |--ztest
+         |--zstreamdump
+      |--+ src/
+         |--+ zfs-0.8.4/
+         |--+ spl-0.8.4/
+
+
+#. Load the new kernel module:
+
+   .. code-block: bash
+
+      sudo modprobe zfs
+
+
+Set up systemd
+==============
+
+You should now have these unit files available. 
+
+.. code-block:: console
+
+   zfs-env-bootfs.service
+   zfs-zed.service
+   zfs-import-cache.service
+   zfs-import-scan.service
+   zfs-mount.service
+   zfs-share.service
+   zfs-volume-wait.service
+  
+
+If you want to use ZFS automatic zpool import and filesystem
+mount services, link the systemd.unit files into /etc and enable them.
+
+.. code-block:: bash
+
+   sudo ln -s ./etc/systemd/system/zfs-import-cache.service /etc/systemd/system/
+   sudo ln -s ./etc/systemd/system/zfs-mount.service /etc/systemd/system/
+
+   systemctl enable zfs-import-cache
+   systemctl enable zfs-import.target
+
+   systemctl enable zfs-mount
+   systemctl enable zfs.target
+
+OpenZFS requires you to explicitly install and enable the services you want. 
```

Comment:
I suggest presenting this information earlier, towards the top of this sections header.